src/core/include/openvino/pass/pattern/op/pattern.hpp

```diff
@@ -104,4 +104,24 @@ class OPENVINO_API Pattern : public Node {
 }  // namespace op
 }  // namespace pattern
 }  // namespace pass
+OPENVINO_API pass::pattern::op::ValuePredicate operator||(const std::function<bool(Output<Node>)>& a,
```

Comment:
why can't we move it to `pass::pattern::op` namespace above?

Reply:
same question for code below.

Reply:
I'm not even sure that operator for arguments `const std::function<bool(Output<Node>)>` will be automatically resolved w/o explicit `using namespace ov`

Reply:
I put these operators into the `ov` namespace to make them more accessible. Most common transformations are in the `ov::pass` namespace, so predicates would be available since `ov` is a parent namespace. As per the plugin transformations -- they have their namespaces, however, they all share `ov` as a parent namespace.
Additionally, I didn't want to override operator and not put it into our namespace to avoid unwanted behavior change for the developers using our product.